sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -376,14 +376,20 @@ trait V2WriteCommand extends Command {
   override lazy val resolved: Boolean = outputResolved
 
   def outputResolved: Boolean = {
-    table.resolved && query.resolved && query.output.size == table.output.size &&
-        query.output.zip(table.output).forall {
-          case (inAttr, outAttr) =>
-            // names and types must match, nullability must be compatible
-            inAttr.name == outAttr.name &&
+    table.resolved && query.resolved && {
+      // When the table schema is empty, skip the schema check.
+      // TODO: introduce a table capability to disable schema check.
+      table.schema.isEmpty || {
```

Comment:
Why is this necessary for removing `SaveMode`? It seems unrelated.

I think that we should add the table capability instead of adding this check. What does this unblock?

Reply:
The noop data source reports its schema as empty, but can accept input data set of any schema.

Previously, `SupportsSaveMode` was a special case, and we skip schema check for it. Now, only file source is a special case,  and schema check is applied to noop data source as well.

Reply:
It sounds like this commit depends on adding a capability to turn off schema validation. That commit should go in first to avoid adding code that we have to remember to revert later.